### PR TITLE
feat: TEP-0097 change in taskrun breakpoints api fields

### PR DESCRIPTION
**Pull request by chengjoey**
# Changes
make TaskBreakpoints struct for taskrun debug.
breakpoint on failure of a step was moved to `breakpoints.onFailure` spec and onFailure breakpoint are now enabled by setting "taskRun.spec.debug.breakpoints.onFailure" to "enabled".

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [x] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) if any changes are user facing, including updates to minimum requirements e.g. Kubernetes version bumps
- [x] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [x] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [x] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
action required: syntax change for taskRun breakpoints (alpha feature). TaskRun breakpoints are now enabled by setting "taskRun.spec.debug.breakpoints.onFailure" to "enabled".
```

### CODE REVIEW

**Comment by chengjoey** [comment]
this pr is splited from [5691](https://github.com/tektoncd/pipeline/pull/5691), only the breakpoint api field has been adjusted

File: pkg/apis/pipeline/v1/taskrun_types.go [hunk 1]
```diff
@@ -105,9 +106,66 @@ const (
...
+	// +optional
+	Breakpoints *TaskBreakpoints `json:"breakpoints,omitempty"`
+}
+
+// TaskBreakpoints defines the breakpoint config for a particular Task
+type TaskBreakpoints struct {
+	// if enabled, pause TaskRun on failure of a step
+	// failed step will not exit
+	// +optional
+	OnFailure   bool     `json:"onFailure,omitempty"`
```
[the remark this answers is not in the document]
Comment (chengjoey) [reply pkg/apis/pipeline/v1/taskrun_types.go:118]: The PR for the relevant type change is [here](https://github.com/tektoncd/community/pull/1003),PTAL！

File: pkg/apis/pipeline/v1/taskrun_types.go [hunk 2]
```diff
@@ -105,9 +106,66 @@ const (
...
+	BeforeSteps []string `json:"beforeSteps,omitempty"`
 	// +optional
 	// +listType=atomic
-	Breakpoint []string `json:"breakpoint,omitempty"`
+	AfterSteps []string `json:"afterSteps,omitempty"`
```
[the remark this answers is not in the document]
Comment (chengjoey) [reply pkg/apis/pipeline/v1/taskrun_types.go]: thanks @lbernick , i have removed `beforeSteps` and `afterSteps` in this PR, and changed commit message and pr description

File: pkg/apis/pipeline/v1/taskrun_types.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (chengjoey) [reply pkg/apis/pipeline/v1/taskrun_types.go]: hi @lbernick , `Onfailure` uses the string type as described in TEP

**Review by lbernick — APPROVED** [review]

**Comment by lbernick** [comment]
Thanks @chengjoey, looks good! Suggested release note: "action required: syntax change for taskRun breakpoints (alpha feature). TaskRun breakpoints are now enabled by setting "taskRun.spec.debug.breakpoints.onFailure" to "enabled"."

File: pkg/apis/pipeline/v1/taskrun_validation.go [hunk 3]
```diff
@@ -212,16 +212,14 @@ func combineParamSpec(p ParamSpec, paramSpecForValidation map[string]ParamSpec)
 	return paramSpecForValidation, nil
 }
 
-// validateDebug
+// validateDebug validates the debug section of the TaskRun.
+// onFailure breakpoint is only allowed to be set as enabled.
```
[the remark this answers is not in the document]
Comment (chengjoey) [reply pkg/apis/pipeline/v1/taskrun_validation.go]: good suggestion! both v1 and v1beta1 have been modified according to this suggestion

**Comment by chengjoey** [comment]
> Thanks @chengjoey, looks good! Suggested release note: "action required: syntax change for taskRun breakpoints (alpha feature). TaskRun breakpoints are now enabled by setting "taskRun.spec.debug.breakpoints.onFailure" to "enabled"."

thanks @lbernick , i have modified release-note by your suggestion

**Comment by chengjoey** [comment]
@afrittoli , you reviewed the previeous PR, could you please review this PR again?🙏

**Comment by jerop** [comment]
@chengjoey please rebase this PR and I'll review it 🙏🏾 

/assign

**Comment by chengjoey** [comment]
> @chengjoey please rebase this PR and I'll review it 🙏🏾
> 
> /assign

thanks @jerop , I've rebased this pr

**Comment by chitrangpatel** [comment]
Thanks @chengjoey! Can you link other PRs in the description? It will help me understand what's been implemented and whats coming next. Here it seems like all the plumbing is good and feeding input to the `entrypointer`. Is the entrypointer logic implemented or upcoming? Thanks!

**Comment by chengjoey** [comment]
> Thanks @chengjoey! Can you link other PRs in the description? It will help me understand what's been implemented and whats coming next. Here it seems like all the plumbing is good and feeding input to the `entrypointer`. Is the entrypointer logic implemented or upcoming? Thanks!
> 
> Otherwise lgtm.

Thanks @chitrangpatel, this PR is split from [5691](https://github.com/tektoncd/pipeline/pull/5691) and make changes to the onFaileure field as [Tep-0097](https://github.com/tektoncd/community/blob/main/teps/0097-breakpoints-for-taskruns-and-pipelineruns.md#breakpoint-on-failure-of-a-taskrun) described, entrypointer logic is upcoming~

**Comment by chitrangpatel** [comment]
> > Thanks @chengjoey! Can you link other PRs in the description? It will help me understand what's been implemented and whats coming next. Here it seems like all the plumbing is good and feeding input to the `entrypointer`. Is the entrypointer logic implemented or upcoming? Thanks!
> > Otherwise lgtm.
> 
> Thanks @chitrangpatel, this PR is split from [5691](https://github.com/tektoncd/pipeline/pull/5691) and make changes to the onFaileure field as [Tep-0097](https://github.com/tektoncd/community/blob/main/teps/0097-breakpoints-for-taskruns-and-pipelineruns.md#breakpoint-on-failure-of-a-taskrun) described, entrypointer logic is upcoming~

Sounds good! Thanks!

**Pull request merged**